mobilizon_reshare/publishers/coordinator.py

```diff
@@ -30,12 +30,10 @@ def build_publishers(
 class PublicationReport:
     status: PublicationStatus
     reason: str
-    publication_id: UUID
```

Comment:
but how do you know which publisher is associated with the report? Now we have a dict that maps the publisher_id to the repo but that information is external to the class and might not be available if the class is used differently

Reply:
IMHO it depends on *what* you do want to know about the publisher after the publication happened. Right now  the only meaningful datum we store about publishers is their name.  If you are a "user" of the database (i.e. a caller of the functions exposed in `mobilizon_reshare.storage.query`) whenever you ask for a `Publication` you also get its publisher.

If we want to store more state of the `AbstractPublisher`implementation that published a given `Publication` for later inspection (like for example a version of the implementation for debugging purposes), we can indeed add some field to the `Publisher` model and to the report.

This way after you call `save_report` on a `PublisherCoordinatorReport` you can reconstruct every information you need whenever you need it, without having to carry around instances of some `AbstractPublisher`.

Reply:
Then maybe we should store the publication here, so we know what Publication the report is about

Reply:
yes but right now there's no point in using a `PublicationReport` without a `PublisherCoordinatorReport`. Since the `PublisherCoordinatorReport` already binds together publication ids and publications I think we should be ok.